store_data.py

```diff
@@ -0,0 +1,341 @@
+                      default="db-n1-standard-2")
+  parser.add_argument("--username", help="username on the host DB",
+                      default="root")
+  parser.add_argument("--password",
+                      help="password for the username on the host DB",
+                      default="password")
+  parser.add_argument("--database", help="name of the database",
+                      default="envoy_stat_db")
+  parser.add_argument("--json_result", help="the JSON result file",
+                      default="result.json")
+  parser.add_argument("--table_name", help=("the table which stores "
+                                            "the benchmarking data"),
+                      default="envoy_stat")
+  parser.add_argument("--project",
+                      help="the project name in Google Cloud.",
+                      default="envoy-ci")
+  parser.add_argument("--logfile",
+                      help="the local log file for this script. New log will be"
+                           "appended to this file.", default="benchmark.log")
+  parser.add_argument("--ownip", help=("the machine's IP from where"
+                                       " the script is being run"),
+                      default="127.0.0.1")
+  parser.add_argument("--envoy_hash",
+                      help="the hash of envoy version",
+                      required=True)
+  parser.add_argument("--runid",
+                      help="the run id of this benchmark",
+                      default="0")
+
+  utils.CreateBooleanArgument(parser, "create_instance",
+                              ("turn on if you want to create"
+                               " a Google Cloud SQL instance"),
+                              create_instance=True)
+
+  utils.CreateBooleanArgument(parser, "create_db",
+                              ("turn on if you want"
+                               " to create the DB"),
+                              create_db=True)
+
+  utils.CreateBooleanArgument(parser, "delete_db",
+                              ("turn on if you want"
+                               " to delete the DB"),
+                              delete_db=True)
+
+  args = parser.parse_args()
+  logfile = open(args.logfile, "ab")
+
+  if args.create_instance:
+    sh_utils.RunGCloudService(["instances", "create", args.db_instance_name,
+                               "--tier", args.tier], args.project,
+                              "sql", logfile)
+    print "Google Cloud SQL Instance {} is created.".format(
+        args.db_instance_name)
+  else:
+    print "DB Instance creation is skipped due to --no-create_instance."
+
+  sh_utils.RunGCloudService(["users", "set-password", args.username, "%",
+                             "--instance", args.db_instance_name, "--password",
+                             args.password], project=args.project,
+                            service="sql", logfile=logfile)
+  print "DB Usernames and passwords are set."
```

Comment:
Wait a sec, can't we dynamically generate the password in the script here and then pass to gcloud and to the MySQLdb connection? That way we never need to put it anywhere persistent like the environment or command-line. That seems the safest.